src/Amalgam/evaluablenode/EvaluableNodeTreeManipulation.cpp

```diff
@@ -1097,79 +1078,44 @@ bool EvaluableNodeTreeManipulation::CollectLabelIndexesFromNormalTree(EvaluableN
 
 		//attempt to put the label in the index
 		auto [_, inserted] = index.insert(std::make_pair(label_sid, tree));
-
-		//if label already exists
-		if(!inserted)
-			return true;
+		collected_all_label_values = inserted;
```

Comment:
Does this do what you want? You will only get the last label's inserted bool...

Reply:
Ah, I see. I misunderstood the logic in the loop.

And nope, I don't think you can bind to an existing var like that.